scipy/linalg/tests/test_decomp_cholesky.py

```diff
@@ -65,6 +67,19 @@ def test_random_complex(self):
             a = dot(c, transpose(conjugate(c)))
             assert_array_almost_equal(cholesky(a, lower=1), c)
 
+    @pytest.mark.xslow
+    def test_int_overflow(self):
+       # regression test for
+       # https://github.com/scipy/scipy/issues/17436
+       # the problem was an int overflow in zeroing out
+       # the unused triangular part
+       n = 47_000
+       np.random.seed(1234)
```

Comment:
Better to reduce the memory usage for the test for the CI

```suggestion
x = np.eye(n, dtype=np.float64, order='F')
x[:4, :4] = np.array([[4,-2,3,-1],[-2,4,-3,1],[3,-3,5,0],[-1,1,0,5]])

cholesky(a, check_finite=False, overwrite_a=True)  # should not segfault
```

Reply:
Got to admit I am not entirely convinced just yet that whatever we come up with will be much better. Happy to be proven wrong though :-).